File: 15. Tree/08HeightofBinaryTree.cpp
#include <bits/stdc++.h> 
using namespace std;

struct node {
    int key;
    node *left;
    node *right;

    node(int k) {
        key = k;
        left = right = NULL;
    }
};

int height(node *root){
    if(root==NULL) return 0;
    else return max( height(root->left) ,  height(root->right)+1 ); 
}

int main() {
    node *root = new node(3);

    root->left = new node(2);
    root->right = new node(4);
    root->right->right = new node(5);
    root->left->left = new node(1);
    
    cout<<"The height of this binary tree is : "<< endl;
    cout<<height(root)<<endl;

    return 0; 
}

File: 15. Tree/Python Visualizer for Trees/08HeightofBinaryTree.py
from visualiser.visualiser import Visualiser as vs

class Node:
    def __init__(self, key):
        self.key = key
        self.left = None
        self.right = None

@vs(node_properties_kwargs={"shape": "record", "color": "#f57542", "style": "filled", "fillcolor": "grey"})
def height(root):
    if root is None:
        return 0
    else:
        return max(height(root.left), height(root.right)) + 1

def create_sample_tree():
    root = Node(3)
    root.left = Node(2)
    root.right = Node(4)
    root.right.right = Node(5)
    root.left.left = Node(1)
    return root

def main():
    root = create_sample_tree()
    print("The height of this binary tree is:")
    print(height(root))
    vs.make_animation("tree_height.gif", delay=1)

if __name__ == "__main__":
    main()
